src/catalog/src/durable/impls/persist.rs

```diff
@@ -357,16 +355,11 @@ impl UnopenedPersistCatalogState {
         while let Some(update) = stream.next().await {
             snapshot.push(update)
         }
-        snapshot
-            .into_iter()
-            .map(|((kind, _unit), ts, diff)| StateUpdate {
-                kind: kind
-                    .expect("kind decoding error")
-                    .try_into()
-                    .expect("kind decoding error"),
-                ts,
-                diff,
-            })
+        snapshot.into_iter().map(|((key, val), ts, diff)| {
+            let (key, ()) = (key.expect("WIP"), val.expect("WIP"));
```

Comment:
Ignore all the error handling, I'll clean that up before I mark this ready for review